### DISCUSSION TITLE
how to login in firebase with php

### DISCUSSION THREAD

**Discussion by harry9216** [kreait/firebase-php#599] [General]
Message: Call to undefined method Kreait\Firebase\Auth::getUserWithEmailAndPassword();

**Comment by jeromegamez** [comment]
I'm not sure what you expect from me or from the SDK: the method you're trying to call does not exist, that's why you get this message.

If you want to sign-in a user with an email and a password, please read the documentation, the method you're looking for is probably this one: https://firebase-php.readthedocs.io/en/5.x/authentication.html#sign-in-with-email-and-password

PS: I find it quite rude to come over and just throw a random message over the fence.

**Reply by harry9216** [reply]
im using sdk 4.17.0  
Type: Error
Message: Call to undefined method Kreait\Firebase\Auth::signInWithEmailAndPassword();

here is my code :
public function user_login(){
$email = $this->input->post('log_email');
$password = md5($this->input->post('log_password'));
$user = $factory->getAuth()->signInWithEmailAndPassword($email,$password);
					if ($user) {
						print_r($user);
					}else{
						echo "failed";
					}
					die();
}

**Reply by jeromegamez** [reply]
For 4.x, please refer to https://firebase-php.readthedocs.io/en/4.x/

You're using a hopelessly outdated version of the SDK, you're using 4.17, the latest 4.x version is 4.44. That's the reason why you don't have all methods available.
